extensions/optaplanner/deployment/src/main/java/io/quarkus/optaplanner/deployment/OptaPlannerProcessor.java

```diff
@@ -80,18 +87,38 @@ void recordSolverFactory(OptaPlannerRecorder recorder, RecorderContext recorderC
                         recorder.initialize(solverConfig, solverManagerConfig)));
     }
 
-    private void applySolverProperties(OptaPlannerRecorder recorder, RecorderContext recorderContext,
+    private boolean applySolverProperties(OptaPlannerRecorder recorder, RecorderContext recorderContext,
             IndexView indexView, SolverConfig solverConfig) {
         if (solverConfig.getScanAnnotatedClassesConfig() != null) {
             throw new IllegalArgumentException("Do not use scanAnnotatedClasses with the Quarkus extension,"
                     + " because the Quarkus extension scans too.\n"
                     + "Maybe delete the scanAnnotatedClasses element in the solver config.");
         }
+
+        // Fail fast if solution class exists but not entity class, and vice versa. If both don't exist, skip extension
+        Pair<Class<?>, Boolean> solutionClassResult = findSolutionClassResult(recorderContext, indexView);
+        List<Class<?>> entityClassList = findEntityClassList(recorderContext, indexView);
+
+        Boolean doesSolutionClassExist = solutionClassResult.getRight();
+        Boolean doesEntityClassExist = !entityClassList.isEmpty();
+
+        if (doesSolutionClassExist && !doesEntityClassExist) {
+            throw new IllegalStateException("A class with a @" + PlanningSolution.class.getSimpleName()
+                    + " annotation was found, but no classes with a @" + PlanningEntity.class.getSimpleName()
+                    + " annotation was found.");
+        } else if (doesEntityClassExist && !doesSolutionClassExist) {
+            throw new IllegalStateException("A class with a @" + PlanningEntity.class.getSimpleName()
+                    + " annotation was found, but no classes with a @" + PlanningSolution.class.getSimpleName()
+                    + " annotation was found.");
+        } else if (!doesSolutionClassExist && !doesEntityClassExist) {
+            return false;
```

Comment:
If neither are found, but the solverConfig has entries, it should not "do nothing". It should continue, at which point it will fail fast.

Reply:
I'll handle this on my end, no need to update pr